src/access/workflows/WithdrawWorkflow.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IAccessPoint} from "../../interfaces/IAccessPoint.sol";
+
+interface IWrappedNativeAsset is IERC20 {
+    function deposit() external payable;
+
+    function withdraw(uint256 amount) external;
+}
+
+contract WithdrawWorkflow {
+    using SafeERC20 for IERC20;
+
+    error NativeWithdrawalFailed();
+
+    function withdrawERC20(IERC20 asset, uint256 amount) external {
+        address owner = _getOwner();
+        asset.safeTransfer({to: owner, value: amount});
```

Comment:
Maybe add events for this? Also for `withdrawNative`

Reply:
This would emit ERC20 events which would be displayed on Etherscan for access point address. I don't see any point of duplicating the events. Also, ETH transfers are quite easy to track on Etherscan as well. Am I missing a point here?

Reply:
nope, i think that's fair!